sklearn/metrics/_classification.py

```diff
@@ -1099,9 +1108,14 @@ def f1_score(
 
         F1 = 2 * (precision * recall) / (precision + recall)
 
-    In the multi-class and multi-label case, this is the average of
-    the F1 score of each class with weighting depending on the ``average``
-    parameter.
+    Support beyond term:`binary` targets is achieved by treating :term:`multiclass`
+    and :term:`multilabel` data as a collection of binary problems, one for each
+    label. For the :term:`binary` case, setting `average='binary'` will return
+    F1 score for `pos_label`. If `average` is not `'binary'`, `pos_label` is ignored
+    and F1 score for both classes are calculated and averaged or both returned (when
+    `average=None`). Similarly, for :term:`multiclass` and :term:`multilabel` targets,
+    F1 score for all `labels` can be returned or use `average` to specify the averaging
+    technique to be used. Use `labels` specify the labels to calculate F1 score for.
```

Comment:
I am always wondering if it is more natural to say "compute" or "calculate" :) (you can change depending what you like better here and in other places)